django/db/models/base.py

```diff
@@ -1861,6 +1861,33 @@ def _check_constraints(cls, databases):
                 )
         return errors
 
+    @classmethod
+    def _warn_unique_constraints(cls, databases):
+        errors = []
+        for db in databases:
+            if not router.allow_migrate_model(db, cls):
+                continue
+            connection = connections[db]
```

Comment:
I would suggest integrating this code into `_check_constraints` above, since then the duplicate code up to here can be shared. The name of that function implies that it is expected to check *all* constraints (I' reading it as "check all constraints", not "do something about check-constraints"). There might even be some code sharing in the error generating (e.g. if the constraint-specific check code sets a `message` and `code` variable, some common code could fill in the rest of the warning (especially if the hints are made equal as I suggest below).

Reply:
Also, it seems that you are not actually calling your new function yet, which should be fixed (or will be automatically fixed if you integrate into `_check_constraints`, which I would prefer).

Reply:
After applying your suggestion, it fails at MariaDB. Seems like MariaDB supports check constraint but doesn't support partial indexing. So, I conclude that the `_check_constraints` method is about "do something about check-constraints", and in the latest commit (testing job ongoing right now) I resplitted back to `_check_unique_constraints` method.

Reply:
My suggestion was to integrate both in the same method, but in that same method look at `supports_table_check_constraints` for `CheckConstraints` as before, and look at `support_partial_indexes` for `UniqueConstraints` with `condition`. I was not trying to suggest that both should only look at `supports_table_check_constraints`, but maybe you did think I did?